### ISSUE TITLE
SC2164 'in case cd fails' false positive for immediate return statements after the `cd`

### ISSUE THREAD

**Issue by eankeen** [koalaman/shellcheck#2135]
#### For bugs
- Rule Id: SC2164 
- version: 0.7.1 (and online) 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/sh

fn() {
	cd folder
	return
}

fn() {
	cd folder
	exit
}

fn() {
	cd folder; return
}

fn() {
	cd folder; exit
}

```

#### Here's what shellcheck currently says:
![image](https://user-images.githubusercontent.com/24364012/106073198-7e7e4c80-6101-11eb-81c0-fc1ecbc8cc5e.png)

#### Here's what I wanted or expected to see:
In case cd fails, the very next statement will either return/exit, which should be enough for `SC2164` not to come up, since it performs the action return/exit, no matter the exit code

**Comment by danisztls** [comment]
Another case where the check is redundant:

```shell
mkdir -p "$_dir_name" && cd "$_dir_name"
```

**Comment by hyperupcall** [comment]
@danisztls I wouldn't say so that in that case because that code is prone to races - and correct error handling is not guaranteed

**Comment by port19x** [comment]
> @danisztls I wouldn't say so that in that case because that code is prone to races - and correct error handling is not guaranteed

How is that snippet prone to races? Doesn't `&&` mandate that `mkdir` finishes before `cd` starts?

**Comment by hyperupcall** [comment]
@port19x Because other processes can run after the `mkdir`, but before the `cd`, that removes `$_dir_name`. With my code, that is not the case.

**Comment by TinCanTech** [comment]
To my _untrained-eye_, the following code appears to function _more_ correctly
```
#!/bin/sh

fn_cd() {
    cd "$folder"
}

folder=temp
fn_cd || exit
```
